Automattic/wp-calypso
# Store: Add stats widget to the store dashboard

This PR implements a stats widget on the dashboard. See p90Yrv-mi-p2.

It is wrapped behind a new feature flag, enabled in development and wpcalypso. 

To Test:
* Run `npm run test-client client/extensions/woocommerce` and make sure all tests pass.
* Go to `http://calypso.localhost:3000/store/:site` on a site with some orders. Test switching the unit (day, week, month). Refresh and make sure the selection persists.
* Verify the different stats/lists load and display correctly.
* If you don’t have a test site with a lot of data, you can force the widget to display another site as an A12.
	* Find a live site by going to Live Tracks view in MC and then search for `calypso_woocommerce_order_received`.  Find an event and expand it. Copy the blogid.
	* Open `client/extensions/woocommerce/app/dashboard/widgets/stats-widget/index.js`
	* Replace `const site = getSelectedSiteWithFallback( state );` with something like `const site = { ID: BLOGID, slug: '', name: 'Test Site' };`.
	* Refresh

<img width="844" alt="screen shot 2018-03-15 at 1 26 56 am" src="https://user-images.githubusercontent.com/689165/37486137-d09c248a-284a-11e8-8aac-29b4b485aa5a.png">

Apologies for the PR size. About 200 lines are tests, and 130 CSS. I could split out the utils and tests (~300 lines) — but I figured it would be easier to review with context.
